src/joystick/windows/SDL_xinputjoystick.c

```diff
@@ -114,116 +113,25 @@ static const char *GetXInputName(const Uint8 userid, BYTE SubType)
-    }
-
-    /* First see if we have a cached entry for this index */
-    if (s_arrXInputDevicePath[userid]) {
-        for (i = 0; i < device_count; i++) {
-            RID_DEVICE_INFO rdi;
-            char devName[128] = { 0 };
-            UINT rdiSize = sizeof(rdi);
-            UINT nameSize = SDL_arraysize(devName);
-
-            rdi.cbSize = sizeof(rdi);
-            if (devices[i].dwType == RIM_TYPEHID &&
-                GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICEINFO, &rdi, &rdiSize) != (UINT)-1 &&
-                GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICENAME, devName, &nameSize) != (UINT)-1) {
-                if (SDL_strcmp(devName, s_arrXInputDevicePath[userid]) == 0) {
-                    *pVID = (Uint16)rdi.hid.dwVendorId;
-                    *pPID = (Uint16)rdi.hid.dwProductId;
-                    *pVersion = (Uint16)rdi.hid.dwVersionNumber;
-                    SDL_free(devices);
-                    return;
-                }
-            }
-        }
+    /* Fixup for Wireless Xbox 360 Controller */
+    if (capabilities.ProductId == 0 && capabilities.Capabilities.Flags & XINPUT_CAPS_WIRELESS) {
+        capabilities.VendorId = USB_VENDOR_MICROSOFT;
+        capabilities.ProductId = USB_PRODUCT_XBOX360_XUSB_CONTROLLER;
     }
 
-    for (i = 0; i < device_count; i++) {
-        RID_DEVICE_INFO rdi;
-        char devName[MAX_PATH] = { 0 };
-        UINT rdiSize = sizeof(rdi);
-        UINT nameSize = SDL_arraysize(devName);
-
-        rdi.cbSize = sizeof(rdi);
-        if (devices[i].dwType == RIM_TYPEHID &&
-            GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICEINFO, &rdi, &rdiSize) != (UINT)-1 &&
-            GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICENAME, devName, &nameSize) != (UINT)-1) {
-#ifdef DEBUG_JOYSTICK
-            SDL_Log("Raw input device: VID = 0x%x, PID = 0x%x, %s\n", rdi.hid.dwVendorId, rdi.hid.dwProductId, devName);
-#endif
-            if (SDL_strstr(devName, "IG_") != NULL) {
-                SDL_bool found = SDL_FALSE;
-                for (j = 0; j < SDL_arraysize(s_arrXInputDevicePath); ++j) {
-                    if (!s_arrXInputDevicePath[j]) {
-                        continue;
-                    }
-                    if (SDL_strcmp(devName, s_arrXInputDevicePath[j]) == 0) {
-                        found = SDL_TRUE;
-                        break;
-                    }
-                }
-                if (found) {
-                    /* We already have this device in our XInput device list */
-                    continue;
-                }
-
-                /* We don't actually know if this is the right device for this
-                 * userid, but we'll record it so we'll at least be consistent
-                 * when the raw device list changes.
-                 */
-                if (rdi.hid.dwVendorId == USB_VENDOR_VALVE &&
-                    rdi.hid.dwProductId == USB_PRODUCT_STEAM_VIRTUAL_GAMEPAD) {
```

Comment:
Note to self: Steam needs to be updated to hook XInputGetCapabilitiesEx() and return the correct VID/PID here.